### ISSUE TITLE
Wallet not syncing [macOS]

### ISSUE THREAD

**Issue by pscott** [nanocurrency/raiblocks#867]
Hi, 
I'm coming here after looking at this reddit post : https://www.reddit.com/r/nanocurrency/comments/8j3j26/nano_classic_wallet_v13_for_windows_not/ .

I'm running on macOS High Sierra 10.13.1, 2.3 GhZ i5 with 8 GB RAM.
My wallet worked fine from December to March this year. 
I got v13 yesterday and got it syncing for 24h+ hours but it wouldn't sync at all. 

I attached the logs, hope it helps.

[log_2018-05-13_19-21-16.76.log](https://github.com/nanocurrency/raiblocks/files/1999617/log_2018-05-13_19-21-16.76.log)
[log_2018-05-13_20-00-17.77.log](https://github.com/nanocurrency/raiblocks/files/1999618/log_2018-05-13_20-00-17.77.log)
[log_2018-05-13_20-39-19.78.log](https://github.com/nanocurrency/raiblocks/files/1999619/log_2018-05-13_20-39-19.78.log)
[log_2018-05-13_21-14-53.79.log](https://github.com/nanocurrency/raiblocks/files/1999620/log_2018-05-13_21-14-53.79.log)
[log_2018-05-14_08-52-17.80.log](https://github.com/nanocurrency/raiblocks/files/1999621/log_2018-05-14_08-52-17.80.log)

**Comment by cryptocode** [comment]
@pscott what's your block counts?

**Comment by pscott** [comment]
That's what I thought, but is has literally been more than 24hours haha

**Comment by Nanoraii** [comment]
Hello,
I'm the one who made the reddit thread mentioned above. I have the same issue as pscott but on Windows and from the thread you can see that there are also other people with it and all at about the same level of synching. 
My block count was 7744627 (1456515) when I gave up and pulled a snapshot. 
So it seems like there is really something wrong at that block count.

**Comment by rotilho** [comment]
Same from here: https://github.com/nanocurrency/raiblocks/issues/823

**Comment by Nanoraii** [comment]
Interesting, also at 7.7 million blocks... I really wonder what happened at this point...

**Comment by Nanoraii** [comment]
I feel like it has something to do with the release of the canary block. 7.7 million blocks were reached on April 11. Exactly when the canary block was released.

**Comment by dchervov** [comment]
I was waiting for two days for v13.0 to synchronize (on Ubuntu), then gave up and ran v12.1 in parallel on another Ubuntu machine - it synchronized from scratch in just 1 hour and 10 minutes.
v13.0 is still stuck at "7746122 (1518204)".

**Comment by justdmitry** [comment]
Same problem with v13.0 node.

Sync stuck at 7,746,299, log full of "Block ... was republished to peers" and "Broadcasting confirm req for block ... to 8 representatives" with same blocks, for example `6872CBA6C17A89AA0C88D1B208A80A8F3D1108F4469EC1BF4AC0D4B9CBAB8BCD`.

Also log contains lines like:

    [2018-05-15 18:23:24.577331]: 338 blocks have been unconfirmed averaging 1049 announcements

And 9k log lines later

    [2018-05-15 18:29:17.635485]: 338 blocks have been unconfirmed averaging 1071 announcements

It seems that node trying to re-confirm same blocks again and again.

Using a snapshot is a wrong way IMHO - blocklattice must be verifiable and sync'able from the ground up. I'll try v12.1...

**Comment by Nanoraii** [comment]
@justdmitry Is that Hash correct? I can't find that block in any block explorer...

**Comment by justdmitry** [comment]
@Nanoraii Here is part of my log, between "338 blocks have been unconfirmed averaging...". I think you will find all 338 blocks here.

[log_2018-05-15_18-23-08.22-3.log](https://github.com/nanocurrency/raiblocks/files/2005753/log_2018-05-15_18-23-08.22-3.log)

**Comment by justdmitry** [comment]
By the way, I backup'ed data-files, installed v12.1 node and started it with v13 database. After 3-5 minutes later it moved to 7,778,986, after 20min - to 7,841,421.

It seems that v13 have some "infinite confirm loop" that v12.1 does not.

Here is log from v12.1: [log_2018-05-15_19-06-20.25-2.log](https://github.com/nanocurrency/raiblocks/files/2005773/log_2018-05-15_19-06-20.25-2.log). I can't find here any block from v13 stuck-log...

**Comment by justdmitry** [comment]
> curl -d '{"action":"unchecked_get", "hash":"6872CBA6C17A89AA0C88D1B208A80A8F3D1108F4469EC1BF4AC0D4B9CBAB8BCD"}' localhost:7076
>> { "contents": "{\n    \"type\": \"receive\",\n    \"previous\": \"12F1F4C94C61A632E7B48548C3E6740E9808CF7EEBC50F4361011D1B99616384\",\n    \"source\": \"A5466412D80BB689B90864C559E46EC1D69438EC301D99F9590331EB81D17CBD\",\n    \"work\": \"119258f02ed5568c\",\n    \"signature\": \"4F8BE5E92E4254FB45BD41961A7D15A0E7B188069E132961B204D5A6739E0E1E20668D739AE0663528FF16EBBECC137B6D58D9D6BA416914481B4C6DAD8B8C0D\"\n}\n" }
>
> https://www.nanode.co/block/A5466412D80BB689B90864C559E46EC1D69438EC301D99F9590331EB81D17CBD

> curl -d '{"action":"unchecked_get", "hash":"12F1F4C94C61A632E7B48548C3E6740E9808CF7EEBC50F4361011D1B99616384"}' localhost:7076
>> { "error": "Unchecked block not found" }

> curl -d '{"action":"block", "hash":"A5466412D80BB689B90864C559E46EC1D69438EC301D99F9590331EB81D17CBD"}' localhost:7076
>> {  "contents": "{\n    \"type\": \"send\",\n    \"previous\": \"9E4776C402CAA1C6986B36BFFA25C0B66D1EBF552CD25AEFA739F42D218646EF\",\n    \"destination\": \"xrb_1tig1rio7iskejqgy6ap75rima35f9mexjazdqqquthmyu48118jiewny7zo\",\n    \"balance\": \"00000000000000000000000000000000\",\n    \"work\": \"e6240bcc83cc666f\",\n    \"signature\": \"F515D40CCC45C34B9DAC9B0628D7B9AE9DDB0D3B4FA979EF5903D19A9295F4749AE630B59E6023D04D7C2A1A7969A35CA56546302B8B2C18C44AE13703C18B0A\"\n}\n" }
> 
> https://www.nanode.co/account/xrb_1tig1rio7iskejqgy6ap75rima35f9mexjazdqqquthmyu48118jiewny7zo - `OKex Rep`

I guess there were some forks in the past, and blocks are still "alive".

**Comment by VenKamikaze** [comment]
Not sure if this would help, but I appeared to get this or a similar issue on the beta nano network. My node was stuck at about 9.9k blocks and wouldn't sync up. I was also trying to create and process a new state 'open' block from this same node and for some reason my blocks were disappearing.

I took a copy of the data.ldb, a relevant log file, and have a copy of the rai_node exec for the version that I was running that was having issues (note: it was a git master compile probably around June30). I would have no issue sending this to the team if it would help?

BTW I "fixed" it by deleting the data.ldb, recompiled rai_node from latest git master, and it sync'd up quick and fine, so it didn't re-occur again. Also, after doing and it syncing, I re-ran the same RPC command that was making my state 'open' block vanish previously, and it now works fine.
